### ISSUE TITLE
Uploader: don’t make the account choice a blocking popup 

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/android#1046]
I have 2 accounts configured in my ownCloud mobile app. There is a blocking account chooser popup every time I want to upload something.

This isn’t great beacuse most of the time I use my personal instead of my company account anyway. So:
- we should directly show the file list, defaulting to the account last used
- put a switcher dropdown in the top bar

Ok @tobiasKaminsky @davivel? :)

**Comment by tobiasKaminsky** [comment]
Good idea.
Then we need a good icon for this @jancborchardt :)

**Comment by jancborchardt** [comment]
Well, the user profile picture from the server, which we also need to use in the Material Design sidebar. Do we have that cc @davivel @DeepDiver1975?

**Comment by AndyScherzinger** [comment]
When my PR for the new drawer/Material design gets merged I can add a account switcher to the drawer. My guess or or more like my expection when uploading a file would then be that it gets uploaded to the "active" account which has been chosen in the drawer. Kind of like the gmail app or other Google apps behave.

**Comment by jancborchardt** [comment]
@davivel @DeepDiver1975 do we have an API to get the profile image? If not, would be good to have that for 8.2

**Comment by AndyScherzinger** [comment]
I second that! :) This would have a nice impact on the mobile UIs since it would be inline with most account integration in other app. As stated by @tobiasKaminsky we would also need a "default" icon/avatar to display in case the user hasn't set an image :)

**Comment by davivel** [comment]
@AndyScherzinger , there is already an account switcher in the code, though it's commented. It's a work started by @tobiasKaminsky. Take it into account, maybe you want to work together in it.

Yes, I would also prefer the switcher in the drawer, not in a dropdown menu.

About the API for the image profile, it doesn't exist AFAIK.

**Comment by AndyScherzinger** [comment]
I also want to put it in the drawer but started out incrementally. So geht the drawer up and running with the existing app and add the account switcher later down the road. 
Am 27.07.2015 20:51 schrieb "David A. Velasco" <notifications@github.com>:@AndyScherzinger , there is already an account switcher in the code, though it's commented. It's a work started by @tobiasKaminsky. Take it into account, maybe you want to work together in it.

Yes, I would also prefer the switcher in the drawer, not in a dropdown menu.

About the API for the image profile, it doesn't exist AFAIK.

**Comment by LukasReschke** [comment]
> @davivel @DeepDiver1975 do we have an API to get the profile image? If not, would be good to have that for 8.2

@rullzer is working on this at https://github.com/owncloud/core/pull/14856

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky interested in a joint venture on the material_toolbar branch? :D

**Comment by jancborchardt** [comment]
@AndyScherzinger @tobiasKaminsky keep in mind though to make small steps. ;) First Material Design pull request, then another separate one for account switcher etc. :)

**Comment by AndyScherzinger** [comment]
@jancborchardt so we will postpone the switcher until the material branch got merged into dev.

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky interested in a joint venture on the material_toolbar branch? :D

On friday I will copy the branch and update my previous account switching system and upload it.

**Comment by AndyScherzinger** [comment]
Cool :)  Maybe you want to also check out the material_design branch. It contains a toolbar implementation but I could get the drawer icon / back arrow switching right yet... After disabling the drawer icon and showing the back arrow I couldn't get the drawer icon back up when navigating back to the root folder...  :(

**Comment by AndyScherzinger** [comment]
I meant couldn't get it right yet

**Comment by jancborchardt** [comment]
@tobiasKaminsky @AndyScherzinger really really pay attention to not put more stuff in https://github.com/owncloud/android/pull/1076 which can be done in separate pull requests. :) I don’t want to hold your efforts back by any means (you’re awesome! :) but past experience has shown that the bigger a pull request gets, the longer it takes to review and get merged – in some cases it even never happens. :\

**Comment by AndyScherzinger** [comment]
sure, point is, @tobiasKaminsky can branch from material_toolbar and let his PR be against this branch, so it is basically a cascade thingy ;)

**Comment by tobiasKaminsky** [comment]
@jancborchardt Yes, exactly like Andy said: I can work on the next step (account switching), but only get it merged after the first PR is merged...

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger I will also look after the material_branch problem.

**Comment by AndyScherzinger** [comment]
Thanks a lot. I will search for the problem in parallel :) 
Am 29.07.2015 21:37 schrieb tobiasKaminsky <notifications@github.com>:@AndyScherzinger I will also look after the material_branch problem.

**Comment by john156** [comment]
:+1:

**Comment by jesmrec** [comment]
Fixed
